extensions/interactions/PencilCodeEditor/PencilCodeEditor.js

```diff
@@ -69,7 +69,25 @@ oppia.directive('oppiaInteractivePencilCodeEditor', [
           });
 
           $scope.reset = function() {
-            pce.setCode($scope.initialCode);
+            $modal.open({
+              templateUrl: 'modals/pencilCodeResetConfirmation',
+              backdrop: 'static',
+              keyboard: false,
+              controller: [
+                '$scope', '$modalInstance', function($scope, $modalInstance) {
+                  $scope.cancel = function() {
+                    $modalInstance.dismiss();
+                  };
+
+                  $scope.resetCode = function() {
+                    $modalInstance.close(true);
+                  };
+                }]
+            }).result.then(function(result) {
+              if (result) {
```

Comment:
I can shift this inside the modal function $scope.resetCode() itself, then we won't need it. Should I ?